src/abstractmcmc.jl

```diff
@@ -41,14 +14,16 @@ struct HMCState{
     TKernel<:AbstractMCMCKernel,
     TAdapt<:Adaptation.AbstractAdaptor,
 }
+    "Random number of the state"
+    rng::Random.AbstractRNG
```

Comment:
The original motivation was indeed reproducibility.
I also agree that this will not cut it as we discussed in the other PR. 
Nonetheless, I would leave it in for this PR until we figure out how deal with it upstream.

Reply:
See the discussion in #314 . I've made the same point as @devmotion ; IMO there's no point in doing this. Moreover, the `rng` is available to `callback`, so if it's about reproducibility, then we just make a callback that saves the state + the `rng`.